lib/ohai/plugins/gce.rb

```diff
@@ -32,13 +45,39 @@ def has_gce_metadata?
     can_socket_connect?(Ohai::Mixin::GCEMetadata::GCE_METADATA_ADDR, 80)
   end
 
+  # return the contents of a file if the file exists
+  # @param path[String] abs path to the file
+  # @return [String] contents of the file if it exists
+  def file_val_if_exists(path)
+    if ::File.exist?(path)
+      ::File.read(path)
+    end
+  end
+
+  # looks at the Manufacturer and Model WMI values to see if they starts with Google.
+  # @return [Boolean] Are the manufacturer and model Google?
+  def has_gce_system_info?
+    if RUBY_PLATFORM =~ /mswin|mingw32|windows/
+      require "wmi-lite/wmi"
+      wmi = WmiLite::Wmi.new
+      computer_system = wmi.first_of("Win32_ComputerSystem")
+      if computer_system["Manufacturer"] =~ /^Google/ && computer_system["Model"] =~ /^Google/ 
+        logger.trace("Plugin GCE: has_gce_system_info? == true")
+        return true
+      end
+    else
+      logger.trace("Plugin GCE: has_gce_system_info? == false")
+      false
+    end
+  end
+
   # Identifies gce
   #
   # === Return
   # true:: If gce can be identified
   # false:: Otherwise
   def looks_like_gce?
-    hint?("gce") || has_gce_metadata?
+    hint?("gce") || has_gce_dmi? || has_gce_system_info? || has_gce_metadata?
```

Comment:
This should probably get refactored to be more like what we're doing in AWS

https://github.com/chef/ohai/blob/master/lib/ohai/plugins/ec2.rb#L108

Then you can just kill off has_gce_metadata?

Also this will make it so that we're only attempting to connect to metadata when we *really* look like gce. That allows us to switch over to the IP vs. name based URL as well.

Reply:
refactored.  This will now only attempt to connect to the metadata server when the system info indicates that this node is on GCE.  Would you like to switch to IP based `GCE_METADATA_ADDR` in the same PR?